libsignal-protocol/src/store/mod.rs

```diff
@@ -0,0 +1,84 @@
+pub(crate) mod identity_key_store;
+pub(crate) mod pre_key_store;
+pub(crate) mod session_store;
+pub(crate) mod signed_pre_key_store;
+
+pub use self::{
+    identity_key_store::IdentityKeyStore,
+    pre_key_store::PreKeyStore,
+    session_store::{SerializedSession, SessionStore},
+    signed_pre_key_store::SignedPreKeyStore,
+};
+
+#[cfg(feature = "in-memory-store")]
+pub mod in_memory_store {
+    use crate::{keys::IdentityKeyPair, Address, InternalError};
+    use std::{borrow::ToOwned, collections::HashMap, sync::Mutex};
+
+    pub struct InMemoryIdentityKeyStore<'a> {
+        trusted_keys: Mutex<HashMap<Address<'a>, Vec<u8>>>,
+        local_registration_id: u32,
+        identity_key_pair: IdentityKeyPair,
+    }
+
+    impl<'a> InMemoryIdentityKeyStore<'a> {
+        pub fn new(
+            identity_key_pair: IdentityKeyPair,
+            local_registration_id: u32,
+        ) -> Self {
+            Self {
+                trusted_keys: Mutex::new(HashMap::new()),
+                local_registration_id,
+                identity_key_pair,
+            }
+        }
+    }
+
+    impl<'a> super::IdentityKeyStore<'a> for InMemoryIdentityKeyStore<'a> {
+        fn identity_key_pair(&self) -> Result<IdentityKeyPair, InternalError> {
+            Ok(self.identity_key_pair.clone())
+        }
+
+        fn local_registration_id(&self) -> Result<u32, InternalError> {
+            Ok(self.local_registration_id)
+        }
+
+        fn save_identity(
+            &self,
+            address: Address<'a>,
```

Comment:
here, `rustc` says some things about lifetimes...
I just added that lifetime here, and also for the `IdentityKeyStore`.

Reply:
See my previous comments.

As part of testing [I made my own `OwnedAddress`](https://github.com/Michael-F-Bryan/libsignal-protocol-rs/blob/851827ef68eb01177027f54ca64d48ebe76024e6/libsignal-protocol/tests/helpers/mod.rs#L155-L159) type to get around the issue that this method is just borrowing a `*const signal_address` from C.